sdk/identity/identity/src/credentials/chainedTokenCredential.ts

```diff
@@ -5,17 +5,33 @@ import { AccessToken, TokenCredential, GetTokenOptions } from "@azure/core-http"
 import { AggregateAuthenticationError, CredentialUnavailable } from "../client/errors";
 import { createSpan } from "../util/tracing";
 import { CanonicalCode } from "@opentelemetry/api";
+import { credentialLogger, CredentialLogger } from "../util/logging";
 
 /**
  * Enables multiple `TokenCredential` implementations to be tried in order
  * until one of the getToken methods returns an access token.
  */
 export class ChainedTokenCredential implements TokenCredential {
+  /**
+   * The name to use during logging
+   */
+  protected credentialName = "ChainedTokenCredential";
+
   /**
    * The message to use when the chained token fails to get a token
    */
   protected UnavailableMessage =
-    "ChainedTokenCredential failed to retrieve a token from the included credentials";
+    "ChainedTokenCredential => failed to retrieve a token from the included credentials";
+
+  private _logger?: CredentialLogger;
+  private get logger(): CredentialLogger {
```

Comment:
I don't like this pattern. Having two properties (`_logger` and `logger`) that are both private, but one is a magic initializing getter seems unnecessarily convoluted and surprising.

It also seems weird that we care to log the wrapper types (e.g. `DefaultAzureCredential`) but we want to pretend that the base class isn't its own layer (`ChainedTokenCredential`) even though `ChainedTokenCredential` can be used directly by external customers.

Is it even useful to log the chained credential layer? Presumably it will always be actual credentials that succeed/fail, and we're just double-logging at this point.

Reply:
The ChainedTokenCredential is it's own beast. We do care to make separate logging here than in the internals, since we want to eventually have a smaller shape of the errors appear in this layer, or at least that's what the Identity team is projecting.

This getter specifically is an attempt to avoid having to do meaningful API changes to this class, but I don't like this approach either.